packages/wonder-blocks-form/src/components/text-field.js

```diff
@@ -73,44 +88,69 @@ export default class TextField extends React.Component<Props, State> {
         disabled: false,
     };
 
+    constructor(props: Props) {
+        super(props);
+        if (props.validation) {
+            // Ensures error is updated on unmounted server-side renders
+            this.state.error = props.validation(props.value) || null;
+        }
+    }
+
     state: State = {
         error: null,
         focused: false,
     };
 
-    handleOnChange: (event: SyntheticInputEvent<HTMLInputElement>) => mixed = (
-        event,
-    ) => {
-        const {validation, onValidation, onChange} = this.props;
-        const newValue = event.target.value;
+    componentDidMount() {
+        this.maybeValidate(this.props.value);
+    }
+
+    maybeValidate: (newValue: string) => void = (newValue) => {
+        const {validation, onValidation} = this.props;
         if (validation) {
-            const maybeError = validation(newValue);
+            const maybeError = validation(newValue) || null;
             this.setState({error: maybeError});
             if (onValidation) {
                 onValidation(maybeError);
             }
         }
+    };
+
+    handleOnChange: (event: SyntheticInputEvent<HTMLInputElement>) => mixed = (
+        event,
+    ) => {
+        const {onChange} = this.props;
+        const newValue = event.target.value;
+        this.maybeValidate(newValue);
         onChange(newValue);
     };
 
     handleOnFocus: (
         event: SyntheticFocusEvent<HTMLInputElement>,
     ) => mixed = () => {
+        const {onFocus} = this.props;
         this.setState({
             focused: true,
         });
+        if (onFocus) {
+            onFocus();
```

Comment:
Oops I think we missed one important part in the implementation spec.

It would be better if `onFocus` returns the `event` handled here, in case the parent component needs to do something with that event.

`onFocus(event);`

Same thing should happen with `onBlur(event)`.

Sorry that I missed that part when I originally wrote the impl. spec.

Reply:
It's all good. I'll made that change and update the implementation spec as well...